Source/WebKit/NetworkProcess/Downloads/Download.h

```diff
@@ -129,10 +136,12 @@ class Download : public IPC::MessageSender, public CanMakeWeakPtr<Download>, pub
 #if PLATFORM(COCOA)
     RetainPtr<NSURLSessionDownloadTask> m_downloadTask;
     RetainPtr<NSProgress> m_progress;
-#endif
 #if HAVE(MODERN_DOWNLOADPROGRESS)
     RetainPtr<NSData> m_bookmarkData;
     RetainPtr<NSURL> m_bookmarkURL;
+    bool m_hasReceivedPlaceholderURL { false };
+    bool m_hasReceivedFinalURL { false };
```

Comment:
These boolean states seem like they shouldn't be necessary with proper use of sendWithAsyncReply, but I'm not sure.